sycl/ReleaseNotes.md

```diff
@@ -1,3 +1,227 @@
+
+## Bug fixes
+### SYCL Compiler
+ - Fixed crash which could happen in corner cases when null attribute created
+   [cec6469]
+ - Fixed crash when lowering `__sycl_alocateLocalMemory` [4960e71]
+ - Fixed workflow for multi-file compilation in AOT mode [a0099a5]
+ - Fixed problem with unbundling from object for device archives for FPGA
+   [25ea6e1]
+ - Do not imply `defaultlib msvcrt` for Linux based driver on Windows
+   [d3dc212d]
+ - Fixed handling of `[[intel::max_global_work_dim()]]` in case of
+   redeclarations [9b615928]
+ - Fixed incorrect diagnostics in the presence of OpenMP [cbec0b5f]
+ - Fixed an issue with incorrect output project report when using `-o` option
+   with FPGA AOT enabled [18ac1723]
+ - Removed restriction that was preventing from applying
+   `[[intel::use_stall_enable_clusters]]` attribute to ANY function [15da879d]
+ - Fixed bugs with recursion in SYCL kernels - diagnostics won't be emitted on
+   using recursion in a discarded branch and in constexpr context [9a9a018c]
+ - Fixed handling of `intel::use_stall_enable_clusters` attribute [06e4ebc7]
+### SYCL Library
+ - Fixed build issue when CUDA 11 is used [f7224f1]
+ - Fixed caching of sub-devices in Level Zero backend[4c34f93]
+ - Fixed requesting of USM memory allocation info on CUDA [691f842]
+ - Fixed [`joint_matrix_mad`](doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc)
+   behaviour to return `A*B+C` instead of assigning the result to `C` [ea59c2b]
+ - Workaround use-case when event isn't waited upon its completion but is
+   queried for its status in an infinite loop [bfef316]
+ - Fixed persistent cache key comparison (esp. when there is `\0` symbol)
+   [3e9ed1d]
+ - Fixed a build issue when `sycl::kernel::get_native` is used [eb17836]
+ - Fixed collisions of helper functions and SPIR-V operations when building with
+   `-O0` or `-O1` [9f2fd98] [c2d6cfa]
+ - [OpenCL] Fixed false-positive assertion trigger when allocation alignment is
+   expected [3351916ad]
+ - Aligned behavior of empty command groups with SYCL2020 [1cf697bd]
+ - Fixed build options handling when they come from different sources
+   [67411472]
+ - Fixed host task CUDA native memory handle [e9cf124b6]
+ - Fixed a memory leak which could happen if a command submission fails
+   [67eac4bd]
+ - Fixed support for math functions `floor/rndd/rndu/rndz/rnde` in ESIMD mode
+   [de694dd8]
+ - Fixed memory allocations for multi-device contexts on Level Zero [f83c9356a]
+ - Renamed `sycl::property::no_init` property to `sycl::property::no_init` in
+   accordance to final SYCL 2020 specification, the old spelling is deprecated
+   [ad46b641]
+ - Use local size specified in `[[sycl::reqd_work_group_size]]` if no local
+   size explicitly passed [0a54bef2]
+
+## API/ABI breakages
+ - None
+
+## Known issues
+ - User-defined functions with the same name and signature (exact match of
+   arguments, return type doesn't matter) as of an OpenCL C built-in
+   function, can lead to Undefined Behavior.
```

Comment:
(if I'm interpreting this correctly)
```suggestion
   arguments, return type doesn't matter), e.g. OpenCL C built-in
   functions, can lead to Undefined Behavior.
```

Reply:
Is the intent here that this problem is specific to collision with OpenCL C built-ins? This is how I interpreted it, and I think the original fits way better if that's the case, and should definitely be reworded if it's not.

Reply:
It's quite old "known issue", my interpretation is that it is UB if a user defines a function which matches an OpenCL C builtin signature(e.g. `size_t get_global_id(size_t)`).
Please, let me know if some rewording is needed here.

Reply:
Ah, I'm getting the intent now. I would suggest to reword the whole item in a more concise way:
```
User-defined functions with the name and signature matching those of any OpenCL C built-in function (i.e. an exact match of arguments, return type doesn't matter) can lead to Undefined Behavior.
```
Another point that could be specified is whether merely _defining_ the function or only calling such a user-defined function can cause UB.

Reply:
> It's quite old "known issue", my interpretation is that it is UB if a user defines a function which matches an OpenCL C builtin signature(e.g. size_t get_global_id(size_t)).

That is not a UB per SYCL spec, that is a bug in our implementation

Reply:
>> That is not a UB per SYCL spec, that is a bug in our implementation

That's right, that is why it is in "known issues" section. Does wording suggested in https://github.com/intel/llvm/pull/4190#discussion_r679069696 sounds OK?

>> Another point that could be specified is whether merely defining the function or only calling such a user-defined function can cause UB.

I think both.

Reply:
> Does wording suggested in #4190 (comment) sounds OK?

LGTM